compiler/GHC/Parser/Lexer.x

```diff
@@ -713,6 +720,10 @@ data Token
   | ITanyclass
   | ITvia
 
+  -- QualifiedDo
+  | ITqdo FastString
```

Comment:
These new constructors (`ITqdo` and `ITqmdo) need to be added to Haddock in `utils/haddock/haddock-api/src/Haddock/Backends/Hyperlinker/Parser.hs:176:3`. However, that is an upstream library, so I'm not sure how we synchronize that. It only warns of unhandled constructors though.